src/Castle.Windsor/Windsor/Installer/FromAssembly.cs

```diff
@@ -115,13 +115,21 @@ public static IWindsorInstaller InDirectory(AssemblyFilter filter, InstallerFact
 		///     <c>MyApp.SuperFeatures.dll</c>, <c>mscorlib.dll</c> and <c>ThirdPartyCompany.UberControls.dll</c> the <c>MyApp.exe</c> and <c>MyApp.SuperFeatures.dll</c> will be scanned for installers, and other
 		///     assemblies will be ignored.
 		/// </remarks>
+#if FEATURE_GETCALLINGASSEMBLY
```

Comment:
After reading the comments through the corefx issues about not wanting to implement `GetCallingAssembly` because of problems with inlining that we haven't actually fully suppressed with `MethodImplOptions.NoInlining` (see quote below from MSDN about tail calls), I think the new methods that require an `Assembly` should be available on both .NET Framework and .NET Core builds, and the old ones using `GetCallingAssembly` should be wrapped in `FEATURE_GETCALLINGASSEMBLY` and kept for compatibility so if users run into problems with them they'll need to switch over to the new ones (if we do get problems in the future we'll want to mark them `[Obsolete]`). I am aware that the team gave in and has added `GetCallingAssembly` to .NET Standard 2.0, I still think this is the best way forward, and will significantly clean up the unit tests so they can all use the same API across both platforms.

> This effect also occurs when M1 executes as a tail call from M2, or when M2 executes as a tail call from M3. You can prevent the JIT compiler from inlining the method that calls GetCallingAssembly, by applying the MethodImplAttribute attribute with the MethodImplOptions.NoInlining flag, but there is no similar mechanism for preventing tail calls. (https://msdn.microsoft.com/en-us/library/system.reflection.assembly.getcallingassembly(v=vs.110).aspx)